app/models/topic_converter.rb

```diff
@@ -66,18 +66,45 @@ def posters
     @posters ||= @topic.posts.where("post_number > 1").distinct.pluck(:user_id)
   end
 
+  def increment_users_post_count
+    update_users_post_count(:increment)
+  end
+
+  def decrement_users_post_count
+    update_users_post_count(:decrement)
+  end
+
+  def update_users_post_count(action)
+    operation = action == :increment ? "+" : "-"
+
+    DB.exec(<<~SQL)
+    UPDATE user_stats
+    SET post_count = post_count #{operation} X.count
+    FROM (
+      SELECT
+        us.user_id,
+        COUNT(*) AS count
+      FROM user_stats us
+      INNER JOIN posts ON posts.topic_id = #{@topic.id.to_i} AND posts.user_id = us.user_id
+      WHERE posts.post_number > 1
+      AND NOT posts.hidden
+      AND posts.post_type = #{Post.types[:regular].to_i}
+      GROUP BY us.user_id
```

Comment:
Similar to above, as with queries, it's nice to have like

```
users_something_query = <<~SQL...
```

so that when I read it I can answer and verify if the statement matches your intent.

Reply:
I added a comment above instead as I couldn't come up with a variable name that can represent what the query is trying to do. Good point though as this definitely improves readability.